java/clients/src/main/java/sleeper/clients/QueryWebSocketClient.java

```diff
@@ -178,12 +186,18 @@ public void onClose(int code, String reason, boolean remote) {
         public void onError(Exception error) {
             messageHandler.onError(error);
         }
+
+        @Override
+        public List<Record> getResults(String queryId) {
+            return messageHandler.getResults(queryId);
+        }
     }
 
     public static class WebSocketMessageHandler {
-        private final Gson serde = new GsonBuilder().create();
+        private static final Gson GSON = new GsonBuilder().setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE).create();
```

Comment:
I don't think we can expect this to deserialise records reliably, since it doesn't know the types of the fields. We could deal with that as a separate issue, since for now we only need this for system tests. There's a danger someone will start using this as a query client before we make it actually work.